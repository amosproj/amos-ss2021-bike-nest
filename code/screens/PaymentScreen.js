import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { ImageBackground, Pressable, StyleSheet, Text, TextInput, View, Image, TouchableWithoutFeedback } from 'react-native';
import { Keyboard } from 'react-native'
import { Dimensions } from "react-native";
import { styles } from '../components/EditPersonalInformation/styles';
import { mainstyles } from '../styles/MainStyles';
import Colors from '../styles/Colors';

var width = Dimensions.get('window').width; //full width
var height = Dimensions.get('window').height; //full height

export default function PaymentScreen({ navigation }) {
  return (
    <View style={mainstyles.container}>
      <ImageBackground source={require('../assets/background/background.png')} style={[styles.backgroundImage, { alignItems: 'center' }]}>
        <View style={{ flex: 0.2, alignContent: 'center', justifyContent: 'center', marginTop: 10 }}>
          <Text style={styles.h1}>Meine Bestellung</Text>
        </View>
        <Text style={mainstyles.h3}>
          <Text style={{position: 'absolute', textAlign:'left'}}>Details                                                 </Text>
          <Text style={{position: 'absolute', textAlign:'right'}}>         Ändern</Text>
        </Text>
        <View style={{ flex: 0.2, justifyContent: 'center' }}> 
        <Text>
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center' }}>1 Slot in BIKE NEST                                          </Text> 
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center' }}>3 Stunden</Text> 
        </Text>
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center' }}>Nürnberg HBF</Text>
          <Text></Text>
          <Text>
          <Image source={require('../assets/payment/clock.png')} />
          <Text style={{ color: Colors.UI_Light_2, alignItems: 'center', justifyContent: 'center' }}>   Reserviert für 30min</Text> 
          </Text>
        </View>
        <View style={{ flex: 0.9, justifyContent: 'center', color: Colors.UI_Light_1 }}>
        <Text>
          <Text style={mainstyles.h3}>Zahlungsmethode                          </Text>
          <Image source={require('../assets/payment/plus.png')}/>
          <Text style={mainstyles.h3}>Hinzufügen</Text>
        </Text>
        <View style={{ flex: 0.6, justifyContent: 'center' }}> 
          <Pressable style={[styles.button, { backgroundColor: '#ffffff' }]}>
            <View style={styles.buttonContent}>
              <Image style={styles.buttonImage, { maxWidth: 150 }} source={require('../assets/payment/Paypal1.png')} />
            </View>
          </Pressable>
          <Pressable style={[styles.button, { backgroundColor: '#ffffff' , borderColor: Colors.UI_Light_2, borderWidth: 1 }]}>
            <View style={styles.buttonContent}>
              <Image style={styles.buttonImage} source={require('../assets/payment/Visa.png')} />
            </View>
          </Pressable>
        </View>
          <Image source={require('../assets/payment/Divider.png')} />
        <View style={{ flex: 0.6, justifyContent: 'center'}}> 
          <Text style={mainstyles.label}>Promocode</Text> 
          <TextInput style={mainstyles.inputFieldHalf}
            placeholder='BIKE NEST'
            secureTextEntry={true}/>
          <Image source={require('../assets/payment/Divider.png')} /> 
        </View>
        </View>
        <View style={{ flex: 0.6, justifyContent: 'center' }}> 
        <Text style={mainstyles.h3}>Übersicht</Text> 
        <Text></Text>
        <Text>
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center' }}>Gesamt exkl. Mwst.                                            </Text> 
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center',  fontWeight: 'bold' }}>50€</Text> 
        </Text>
        <Text>
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center' }}>Mwst. 19%                                                          </Text> 
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center',  fontWeight: 'bold' }}>9,50€</Text> 
        </Text>
        <Text>
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center' }}>Rabatt                                                                     </Text> 
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center',  fontWeight: 'bold' }}>-5€</Text> 
        </Text>
          <Text></Text>
          <Image source={require('../assets/payment/Line.png')} />
          <Text>
          <Text style={{ color: '#000000', alignItems: 'center', justifyContent: 'center' }}>Gesamt                                                            </Text> 
          <Text style={{ color: Colors.UI_Light_2, alignItems: 'center', justifyContent: 'center',  fontWeight: 'bold' }}>54,50€</Text> 
          </Text>
          <Text></Text>
          <Text style={{marginLeft: 120, margin: 10,}}>
          <Text style={mainstyles.h3}>Jetzt kostenpflichtig Bestellen  </Text> 
          <Image source={require('../assets/payment/mail-send.png')} />
          </Text>
        </View>
        
        
        <StatusBar style="auto" />
        </ImageBackground>
    </View>
  );
}

// const styles = StyleSheet.create({
//   container: {
//     flex: 1,
//     alignItems: 'center',
//   },
//   backgroundImage: {
//   },
//   h1: {
//     position: 'absolute',
//     left: 50,
//     fontSize: 28,
//     fontWeight: 'bold',
//     alignItems: 'center'
//   },
//   inputField: {
//     width: 328,
//     height: 55,
//     color: '#BDBDBD',
//     backgroundColor: Colors.UI_Light_4,
//     borderRadius: 15,
//     margin: 9,
//     paddingLeft: 12,
//   },
//   button: {
//     alignItems: 'center',
//     justifyContent: 'space-evenly',
//     elevation: 3,
//     width: 328,
//     height: 55,
//     borderRadius: 38,
//     margin: 9,
//     backgroundColor: Colors.UI_Light_2
//   },
//   buttonContent: {
//     flexDirection: 'row',
//     justifyContent: 'space-evenly',
//     width: '100%'
//   },
//   buttonText: {
//     alignSelf: 'center',
//     fontSize: 14,
//     color: Colors.UI_Light_4,
//   },
//   buttonImage: {
//     alignSelf: 'center'
//   }
// });
