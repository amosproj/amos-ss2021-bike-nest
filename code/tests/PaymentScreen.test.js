import React from 'react';
import renderer from 'react-test-renderer';
import PaymentScreen from '../screens/PaymentScreen';

it('renders correctly', () => {
    const params = { state: jest.fn(), name: jest.fn(), slots: jest.fn(), time: jest.fn() };
    const tree = renderer
        .create(<PaymentScreen route={{ params }} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});