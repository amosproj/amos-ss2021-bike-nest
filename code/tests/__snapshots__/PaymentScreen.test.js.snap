// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000090",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "elevation": 3,
            "height": 184,
            "justifyContent": "space-between",
            "padding": 18,
            "width": 288,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "300",
            }
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F28D35",
              "borderRadius": 38,
              "elevation": 3,
              "height": 55,
              "justifyContent": "center",
              "margin": 9,
              "opacity": 1,
              "padding": 18,
              "width": 128,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              Ok
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignSelf": "auto",
            "flex": 1,
            "justifyContent": "space-evenly",
            "marginTop": 30,
            "padding": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 27,
                "fontWeight": "400",
                "letterSpacing": 2,
              }
            }
          >
            Meine Reservierung 
            <Image
              onPress={[Function]}
              source={1}
            />
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "300",
              }
            }
          >
             Details 
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "300",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
             Ändern 
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            

             im BIKE NEST 
            

             
            

          </Text>
          <Text
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
             
             
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "margin": 10,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                },
                Object {
                  "color": "#F28D35",
                },
              ]
            }
          >
            Reserviert für 30min
          </Text>
        </View>
        <Image
          source={1}
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": -15,
              "marginTop": 10,
              "width": 500,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "300",
              }
            }
          >
             Zahlungsmethode
            

          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "maxHeight": 50,
                    "maxWidth": 45,
                    "resizeMode": "contain",
                  },
                ]
              }
            />
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "maxHeight": 50,
                    "maxWidth": 35,
                    "resizeMode": "contain",
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
            ]
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="IBAN"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#F2F2F2",
                "borderRadius": 15,
                "color": "#333333",
                "height": 55,
                "margin": 9,
                "paddingLeft": 12,
                "width": 328,
              }
            }
            underlineColorAndroid="transparent"
          />
          <Text>
            Mit der Reservierung bestätige Ich, dass meine Kontodaten für zukünftige Zahlungen von BIKENEST belastet werden dürfen.*
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 20,
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "margin": 8,
                }
              }
            >
              SEPA-Lastschriftmandat akzeptieren
            </Text>
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "margin": 5,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID="checkbox"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <Image
          source={1}
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": -15,
              "marginTop": 10,
              "width": 500,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
              Object {
                "marginBottom": 10,
                "marginTop": 10,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "300",
              }
            }
          >
            Geschätzter Preis
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "300",
                },
                Object {
                  "color": "#F28D35",
                  "fontWeight": "bold",
                },
              ]
            }
          >
             ~
            € 
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              Object {
                "justifyContent": "flex-end",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "300",
              }
            }
          >
            Jetzt Reservieren
          </Text>
          <Image
            source={1}
            style={
              Object {
                "margin": 10,
              }
            }
          />
        </View>
        <Text>
          *Innerhalb von 14 Tagen können sie diese Einwilligung zurück ziehen. Schreiben Sie dafür unserem Kundenservice.
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Profil
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          (un)lock
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Navigation
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Verlauf
        </Text>
      </View>
    </View>
  </View>
</View>
`;
